modules/users/server/controllers/users.js

```diff
@@ -347,7 +347,7 @@ module.exports = function(System) {
     var email = req.query.email;
     User.findOne({email: email}, function(err, user) {
       if (err || !user) {
-
+        return json.unhappy({message: 'Email not found'}, res);
```

Comment:
Fix: Needed an error response here, this is what caused the server to hang on this request.